presto-hive-common/src/main/java/com/facebook/presto/hive/zorder/ZOrder.java

```diff
@@ -147,6 +157,15 @@ public int encodeToInteger(List<Integer> input)
         int[] output = new int[encodingBits.size()];
 
         int bitIndex = totalBitLength - 1;
+        for (int index = 0; index < output.length; index++) {
+            byte signBit = (byte) ((address[bitIndex >> 3] >> (bitIndex & 7)) & 1);
+            if (signBit == 0) {
+                output[index] -= (1 << encodingBits.get(index));
+            }
+            bitIndex--;
+        }
```

Comment:
I don't understand what we are doing there. I expect that we do the regular decoding, then we revert the sign bits, but here we do something before the regular decoding. Please enlighten me here.

Reply:
That piece of code sets the correct sign bits for the output values. It does not matter whether it is put before or after the decoding of the non-signed bits; it works the same.